api/src/main/java/io/grpc/ClientCall.java

```diff
@@ -138,6 +138,9 @@ public void onMessage(T message) {}
      * An additional block of trailer metadata may be received at the end of the call from the
      * server. An empty {@link Metadata} object is passed if no trailers are received.
      *
+     * <p>This method should not throw. Implementations should be extra-careful about exceptions
```

Comment:
Maybe something akin to: "If this method throws, there is no way to be notified of the exception. Implementations should therefore be careful of exceptions which can accidentally leak resources."

Not wild about my suggested language, but the "hang the call" is too much of a stub concept.